CONTRIBUTING.md

```diff
@@ -119,16 +119,23 @@ Before you submit your pull request consider the following guidelines:
 
 That's it! Thank you for your contribution!
 
-#### <a name="testing"></a> Testing approach
+#### <a name="testing"></a> Testing 
 
-Use the [BDD style](http://guide.agilealliance.org/guide/bdd.html) of testing.
-Tests should be organised around scenarios rather than rigidly around class
-structure and method names, for example "Creating an application" rather than
+Test are organised around scenarios ([BDD style](http://guide.agilealliance.org/guide/bdd.html)) and/or single units ([Unit Tests](http://guide.agilealliance.org/guide/unittest.html)) of work. In the best case we have unit test to verify that every single unit is working as expected and scenarios to ensure that they all play well together.
+
+Keep all tests short, clean, and descriptive. Aim for high code coverage, but don't
+worry about achieving 100%. Try to keep tests flexible. Test the
+interface, not the implementation. Make sure that tests can be run in any order and 
+that tests consistently return the same result. No random numbers and such.  Aim to ensure that tests run quickly to keep the feedback loop tight.
+
+##### Units 
+
+Unit tests should only test a single unit of work, therefor any dependency should be mocked or stubbed.
```

Comment:
also add somewhere in the paragraph: 

The filename MUST match the component being tested, for example `AboutModalComponent.test.js`.